projects/igniteui-angular/src/lib/grids/toolbar/grid-toolbar-advanced-filtering.component.ts

```diff
@@ -50,6 +50,30 @@ export class IgxGridToolbarAdvancedFilteringComponent implements AfterViewInit {
         this.grid?.advancedFilteringExpressionsTreeChange.subscribe(filteringTree => {
             this.numberOfColumns = this.extractUniqueFieldNamesFromFilterTree(filteringTree).length;
         });
+        this.grid?.filteringExpressionsTreeChange.subscribe(() => {
+            setTimeout(() => {
+                if (this.grid.advancedFilteringExpressionsTree) {
+                    const dataKeys = Object.keys(this.grid.data[0]);
```

Comment:
why is this code necessary to be wrapped in setTimeout and even have another one inside this logic on row 69?

Reply:
I used a custom sample to check different scenarios, and the setTimeout functions were part of testing one specific scenario. They were inadvertently left in the code from the large amount of testing done. Thank you for pointing this out;.I’ll remove them immediately.